plugins/janus_sip.c

```diff
@@ -2986,37 +3064,54 @@ static void *janus_sip_handler(void *data) {
 			}
 
 			session->account.registration_status = janus_sip_registration_status_registering;
-			if(!refresh && session->stack == NULL) {
+			if (!refresh) {
 				/* Start the thread first */
 				GError *error = NULL;
 				char tname[16];
 				g_snprintf(tname, sizeof(tname), "sip %s", session->account.username);
 				janus_refcount_increase(&session->ref);
 				g_thread_try_new(tname, janus_sip_sofia_thread, session, &error);
 				if(error != NULL) {
-					janus_refcount_decrease(&session->ref);
 					JANUS_LOG(LOG_ERR, "Got error %d (%s) trying to launch the SIP Sofia thread...\n",
 						error->code, error->message ? error->message : "??");
 					error_code = JANUS_SIP_ERROR_UNKNOWN_ERROR;
 					g_snprintf(error_cause, 512, "Got error %d (%s) trying to launch the SIP Sofia thread",
 						error->code, error->message ? error->message : "??");
 					g_error_free(error);
+					janus_refcount_decrease(&session->ref);
```

Comment:
Why have you moved this?

Reply:
I have a slightly different log line here. I'll return janus_refcount_decrease to the previous place.